### PR TITLE
[WIP] Revise aggregate_files behavior in read_parquet

### LINKED ISSUE #8829: dask DataFrame.read_parquet doesn't preserve the paths order
**What happened**:
Dask doesn't honor the order of files listed in the path

**What you expected to happen**:
I would have expected the partition ordering to be consistent with list of paths in read_parquet. Seems like the input path is automatically sorted internally without choice. 

**Minimal Complete Verifiable Example**:

```python
import dask.dataframe as dd
import numpy as np
df1=pd.DataFrame({'i': [1, 1], 'A': [1.0, 2.0], 'B': [11.0, 12.0]})
df1.to_parquet("df1.parquet")
df2=pd.DataFrame({'i': [0, 0], 'A':  [3.0, 4.0], 'B': [13.0, 14.0]})
df2.to_parquet("df2.parquet")
forward_df = dd.read_parquet(["df1.parquet", "df2.parquet"], engine='pyarrow')
reverse_df = dd.read_parquet(["df2.parquet", "df1.parquet"], engine='pyarrow')
forward_df.npartitions  # 2
reverse_df.npartitions  # 2

forward_df.get_partition(0).compute() # returns df1
forward_df.get_partition(0).compute() # Also returns df1. Expected to return df2


```

**Anything else we need to know?**:

**Environment**:

- Dask version:
- Python version:
- Operating System:
- Install method (conda, pip, source):

<details>
<summary>Cluster Dump State:</summary>

</details>

### LINKED ISSUE #9043: Should we deprecate chunksize from read_parquet?
Follow-up to recent issues like https://github.com/dask/dask/issues/8937

Recent PRs have simplified the `read_parquet` API a bit, but the code is still vast. After some careful consideration, I'd like to suggest that we deprecate the `chunksize` option.

Why to deprecate `chunksize`:
- No one seems to be using it (Please do correct me if I'm wrong!)
- The current implementation is not very efficient (and is extremely slow on remote file systems)
- There is no case where this feature is **required** for stability (probably why it is never used)

cc @jcrist

### LINKED ISSUE #9051: Should we deprecate aggregate_files from read_parquet?
This issue is similar to #8937 (already done) and #9043 in the sense that it aims to remove unnecessary (and rarely-utilized) options from `dd.read_parquet`.

TLDR: I’d like to propose that we deprecate the `aggregate_files` argument from `dd.read_parquet`.

Although I do believe there is strong motivation for a file-aggregation feature (especially for hive/directory-partitioned datasets), the current implementation of this feature actually aggregates row-groups (rather than files), which is extremely inefficient at scale (or on remote storage). This implementation “snafu” is my own fault. However, rather than directly changing the current behavior, I suggest that we simply remove the option altogether.  Removing both `aggregate_files` and `chunksize` (see #9043) should allow us to cut out a lot of unnecessarily-complex core/engine code and reduce general maintenance burden.

In the future, we may wish to re-introduce an `aggregate_files`-like feature, but that (simpler) feature should be designed to aggregate full files (rather than arbitrary row-groups). Users (or down-stream libraries) that need more flexibility than simple “per-row-groups” or “per-files” partitioning should be able to feed their own custom logic into the new `from_map` API.

### PR DESCRIPTION
**Pull request by rjzamora**
Closes #9043
Closes #9051 
Closes #8829

This PR (mostly) preserves the existing `chunksize`/`aggregate_files` options in `read_parquet` by adding two new arguments:

- `sort_input_paths` (default `True`): Whether or not Dask should re-order the files in the dataset to use "natural" ordering. Note that this new feature *could* be added in a separate PR, but I wanted to make sure that the new design allows for such an argument to exist.
- `partition_boundary` (default `None`): A list of directory-partitioned-column names that must match for two or more files to be aggregated into the same output partition. Files within the same "partition boundary" are said to belong to the same "file group."  The engines are always allowed to reorder paths by file group to improve file-aggregation behavior (even if sort_input_paths=False`).  Note that I added this option in order to drop support for `str` arguments to `aggregate_files` in favor of `int` support (explained below).

In addition to these new arguments, I also modified the existing `aggregate_files` argument to only accept `bool` or `int` types. That is, the `aggregate_files` option is now the "file equivalent" of `split_row_groups`. Specifying `aggregate_files=100` means that 100 files from the same file group may be aggregated into the same output partition.

The most important result of this PR is likely the support for `aggregate_files=<int>` (in combination with `partition_boundary=`). For example. In `main`, one would need to use `chunksize` (or `split_row_groups`) with `aggregate_files="year"` to read in a single large partition for each distinct year in a partitioned NYC-taxi dataset:

```python
import dask.dataframe as dd

ddf = dd.read_parquet(
    "s3://ursa-labs-taxi-data/",
    engine="pyarrow",
    storage_options={"anon": True},
    dataset={
        "partitioning": ["year", "month"],
        "partition_base_dir": "ursa-labs-taxi-data",
    },
    chunksize="6GB",
    aggregate_files="year",
)
# (takes several seconds on my workstation - but scales poorly with the total number of row groups)
```

However, with this branch, file aggregation *can* be much faster if the desired number of files per partition is known apriori:

```python
ddf = dd.read_parquet(
    "s3://ursa-labs-taxi-data/",
    engine="pyarrow",
    storage_options={"anon": True},
    dataset={
        "partitioning": ["year", "month"],
        "partition_base_dir": "ursa-labs-taxi-data",
    },
    partition_boundary=["year"],
    aggregate_files=12,  # (or larger)
)
# (takes less than 1s on my workstation)
```

### CODE REVIEW

**Comment by charlesbluca** [comment]
@rjzamora bumping this - are there any blockers here or is this ready for review?
